models/src/main/scala/coop/rchain/models/either/EitherAnyHelper.scala

```diff
@@ -0,0 +1,39 @@
+package coop.rchain.models.either
+
+import coop.rchain.either.EitherAny
+import coop.rchain.models.StacksafeMessage
+
+import scalapb._
+
+object EitherAnyHelper {
+  def pack[A <: StacksafeMessage[A]](generatedMessage: A): EitherAny =
+    pack(generatedMessage, "type.rchain.coop/")
+
+  def pack[A <: StacksafeMessage[A]](
+      generatedMessage: A,
+      urlPrefix: String
+  ): EitherAny =
+    EitherAny(
+      typeUrl =
+        if (urlPrefix.endsWith("/"))
+          urlPrefix + generatedMessage.companion.scalaDescriptor.fullName
+        else
+          urlPrefix + "/" + generatedMessage.companion.scalaDescriptor.fullName,
+      value = generatedMessage.toByteString
+    )
+
+  private def typeNameFromTypeUrl(typeUrl: String): String =
+    typeUrl.split("/").lastOption.getOrElse(typeUrl)
+
+  def is[A <: StacksafeMessage[A]](
+      eitherAny: EitherAny
+  )(implicit cmp: GeneratedMessageCompanion[A]): Boolean =
+    typeNameFromTypeUrl(eitherAny.typeUrl) == cmp.scalaDescriptor.fullName
+
+  def unpack[A <: StacksafeMessage[A]](
+      eitherAny: EitherAny
+  )(implicit cmp: GeneratedMessageCompanion[A]): A = {
+    require(is[A](eitherAny), s"Type of the EitherAny message does not match the given class.")
```

Comment:
Does this `require` halt the node if it fails? Does that mean I can bring down a node by sending it a bad GRPC message?

Reply:
@dckc I don't think so. It's implemented the same way in https://github.com/scalapb/ScalaPB/blob/master/scalapb-runtime/shared/src/main/scala/scalapb/AnyMethods.scala#L13
If you send a bad message to a grpc service then your request will fail. This is true for any kind of grpc service and message.

Reply:
Docs for require say:
> Tests an expression, throwing an IllegalArgumentException if false. -- [predefs](https://www.scala-lang.org/api/current/scala/Predef$.html)

Where is the `IllegalArgumentException` caught? I don't see it. What does the caller get in this case?